### PR TITLE
Fix web root for WebApplication

### LINKED ISSUE #32415: Web root is incorrect when using new minimal hosting API (WebApplication.CreateBuilder)
When using the new minimal hosting API (e.g. `WebApplication.CreateBuilder`) the web root folder is incorrectly set resulting in static files not working.

Issue seems to be here somewhere as it looks like it's trying to combine a path that's already been combined in the constructor:

https://github.com/dotnet/aspnetcore/blob/c9fa4e173426eb39a02109a8344ec9a38c74964d/src/DefaultBuilder/src/WebHostEnvironment.cs#L68

@halter73 @davidfowl 

Found in version 6.0.100-preview.5.21254.10

### PR DESCRIPTION
**Pull request by halter73**
- Use WebRootKey instead of ContentRootKey to populate WebRootPath

Fixes #32415

### CODE REVIEW

File: src/DefaultBuilder/src/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Runtime.CompilerServices;
+
+[assembly: InternalsVisibleTo("Microsoft.AspNetCore.Tests, PublicKey=0024000004800000940000000602000000240000525341310004000001000100f33a29044fa9d740c9b3213a93e57c84b472c84e0b8a0e1ae48e67a9f8f6de9d5f7f3d52ac23e48ac51801f1dc950abe901da34d2a9e3baadb141a17c77ef3c565dd5ee5054b91cf63bb3c6ab83f72ab3aafe93d0fc3c2348b764fafb0b1c0733de51459aeab46580384bf9d74c4e28164b7cde247f891ba07891c9d872ad2bb")]
```
Comment (davidfowl) [inline src/DefaultBuilder/src/Properties/AssemblyInfo.cs:6]: Don't we have a better way to do this from the csproj now?

**Comment by davidfowl** [comment]
@SteveSandersonMS @javiercn @captainsafia @pranavkm  the blazor performance tests are broken

File: src/DefaultBuilder/test/Microsoft.AspNetCore.Tests/WebHostEnvironmentTests.cs [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Tests
+{
+    public class WebHostEnvironmentTests
```
Comment (davidfowl) [inline src/DefaultBuilder/test/Microsoft.AspNetCore.Tests/WebHostEnvironmentTests.cs:14]: Is there a test making sure static files works?

File: src/DefaultBuilder/samples/SampleApp/Program.cs [hunk 3]
```diff
@@ -15,13 +14,14 @@ namespace SampleApp
...
     public class Program
     {
-        public static async Task Main(string[] args)
+        public static void Main(string[] args)
         {
-            await using var webApp = WebApplication.Create(args);
+            var app = WebApplication.Create(args);
 
-            webApp.MapGet("/", (Func<string>)(() => "Hello, World!"));
+            app.UseStaticFiles();
```
Comment (davidfowl) [inline src/DefaultBuilder/samples/SampleApp/Program.cs:21]: I guess you manually tested this?

Comment (halter73) [reply src/DefaultBuilder/samples/SampleApp/Program.cs:21]: Yes. I can write a manual test. There aren't any e2e tests in DefaultBuilder that don't hit the network stack so I was a little lazy and just went with the unit tests and the sample.

**Comment by SteveSandersonMS** [comment]
@davidfowl 

> the blazor performance tests are broken

Can you clarify which ones? The WebAssembly benchmarks still seem to be working fine. And do you mean they are broken already, or just after this PR? The build on CI doesn't seem to indicate that this PR breaks any perf tests.

**Comment by pranavkm** [comment]
It’s the razor compiler issue with generic type parameters.

**Comment by davidfowl** [comment]
> It’s the razor compiler issue with generic constraints.

Is it being fixed? Can these tests be skipped? Is it flaky? We want to merge this PR 😄

**Comment by SteveSandersonMS** [comment]
Yes, it was quarantined yesterday by @captainsafia: https://github.com/dotnet/aspnetcore/pull/32593

**Review by davidfowl — APPROVED** [review]

**Comment by DamianEdwards** [comment]
🥳

**Pull request merged**